### PR TITLE
Don't emit memcpy's for zero-sized types

### PR DESCRIPTION
**Pull request by dotdash**
r? @eddyb

### CODE REVIEW

File: src/librustc_trans/trans/base.rs [hunk 1]
```diff
@@ -956,6 +956,10 @@ pub fn memcpy_ty<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
                              t: Ty<'tcx>) {
     let _icx = push_ctxt("memcpy_ty");
     let ccx = bcx.ccx();
+    if type_is_zero_size(bcx.ccx(), t) {
```
Comment (ranma42) [inline src/librustc_trans/trans/base.rs]: Wouldn't `ccx` be cleaner than `bcx.ccx()` ?

Comment (dotdash) [inline src/librustc_trans/trans/base.rs]: Right, copy-pasted that line, will fix.

**Pull request merged**